src/my_functions_product1.py

```diff
@@ -0,0 +1,721 @@
+   sensitive_field = input_from_user(input_str, warn_str, result_str, possible_set)
+    
+    ### Inputting user preference for Sensitive 0-1 classes
+   possible_set = set(your_data_df[sensitive_field].unique())
+
+
+   print('unique values from the sensitive class:{}'.format(possible_set))
+   print('Please enter two different classes in the sensisitve field: ')
+
+   input_str = 'Enter class 0:'
+   warn_str = 'Please enter a valid class within sensitive field'
+   result_str = "Your sensitive attribute class 0 is "
+   sensitive_class0 = input_from_user(input_str, warn_str, result_str, possible_set)
+
+   possible_set.remove(sensitive_class0)
+
+   input_str = 'Enter class 1:'
+   warn_str = 'Please enter a valid class within sensitive field'
+   result_str = "Your sensitive attribute class 1 is "
+   sensitive_class1 = input_from_user(input_str, warn_str, result_str, possible_set)
+    
+    ### Do you want to omit any column from your data?
+   possible_set = ['Y','N']
+   input_str = 'Do you want to omit any column from your data? Please enter Y or N:'
+   warn_str = 'Please enter Y or N'
+   result_str = "Do you have any column for your current prediction? You entered"
+   fields_to_delete = input_from_user(input_str, warn_str, result_str, possible_set)
+    
+   remaining_columns = all_columns.copy()
+   remaining_columns.remove(target_field)
+
+   if cprediction_field_exists == 'Y':
+       remaining_columns.remove(cprediction_field) 
+
+   columns_can_omitted = remaining_columns
+    
+   not_to_consider_fields = deleting_columns(fields_to_delete,columns_can_omitted)
+
+   return cprediction_field_exists,cprediction_field,target_field, sensitive_field, your_data_df, not_to_consider_fields,sensitive_class0,sensitive_class1,target_label0,target_label1
+#==============================================================================
+   
+def processing_data_frame(output_field, sensitive_field, your_data_df, not_to_consider_fields,class0,class1):
+    
+    #drop columns
+    for item in not_to_consider_fields:
+        print(item)
+        print(sensitive_field)
+        if item != sensitive_field:
+            your_data_df = your_data_df.drop(item, axis=1)
+            #print(item,'----',sensitive_field)
+           # print(item,your_data_df.columns)
+    
+    #check for missing values, do sth with them
+    your_data_df = your_data_df.dropna()
+    
+    # only keep the row with sensitive field equal to class1, and class2
+    your_data_df = your_data_df.loc[lambda df: df[sensitive_field].isin([class0, class1])]
+
+    #create Y
+    Y_df = your_data_df[output_field]
+    
+    #create Z
+    Z_df = your_data_df[sensitive_field]
+    
+    #create X
+    X_df = your_data_df.copy()
+    #print(X_df.shape)
+    X_df = X_df.drop(output_field, axis=1)
+    #print(X_df.shape)
+
+
+    if sensitive_field in not_to_consider_fields:
+        X_df = X_df.drop(sensitive_field, axis=1)
+        
```

Comment:
Sorry, aren't you handling this inside the for loop (if item != sensitive_field). Also what would happen if sensitive_field is a list with item as one of its elements?